extensions/ql-vscode/src/cli.ts

```diff
@@ -131,6 +131,12 @@ export class CodeQLCliServer implements Disposable {
    */
   private static CLI_VERSION_WITH_LANGUAGE = new SemVer('2.4.1');
 
+  /**
+   * CLI version where `codeql resolve upgrades` supports
+   * the `--allow-downgrades` flag
+   */
+  private static CLI_VERSION_WITH_DOWNGRADES = new SemVer('2.4.4');
```

Comment:
Not necessary for this PR, but we should think of a better way of dealing with feature flags since I can only imagine this is going to get worse.